homeassistant/components/squeezebox/config_flow.py

```diff
@@ -0,0 +1,216 @@
+
+        discovery_task = self.hass.async_create_task(
+            async_discover(_discovery_callback)
+        )
+
+        await discovery_event.wait()
+        discovery_task.cancel()  # stop searching as soon as we find server
+
+        # update with suggested values from discovery
+        self.data_schema = _base_schema(self.discovery_info)
+
+    async def _validate_input(self, data):
+        """
+        Validate the user input allows us to connect.
+
+        Retrieve unique id and abort if already configured.
+        """
+        server = Server(
+            async_get_clientsession(self.hass),
+            data[CONF_HOST],
+            data[CONF_PORT],
+            data.get(CONF_USERNAME),
+            data.get(CONF_PASSWORD),
+        )
+
+        try:
+            status = await server.async_query("serverstatus")
+            if not status:
+                if server.http_status == HTTP_UNAUTHORIZED:
+                    return "invalid_auth"
+                return "cannot_connect"
+        except Exception:  # pylint: disable=broad-except
+            return "unknown"
+
+        if "uuid" in status:
+            await self.async_set_unique_id(status["uuid"])
+            self._abort_if_unique_id_configured()
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input and CONF_HOST in user_input:
+            # update with host provided by user
+            self.data_schema = _base_schema(user_input)
+            return await self.async_step_edit()
+
+        # no host specified, see if we can discover an unconfigured LMS server
+        try:
+            await asyncio.wait_for(self._discover(), timeout=TIMEOUT)
+            return await self.async_step_edit()
+        except asyncio.TimeoutError:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema({vol.Optional(CONF_HOST): str}),
+                errors={"base": "no_server_found"},
+            )
+
+        # display the form
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema({vol.Optional(CONF_HOST): str}),
+            errors=errors,
+        )
+
+    async def async_step_edit(self, user_input=None):
+        """Edit a discovered or manually inputted server."""
+        errors = {}
+        if user_input:
+            error = await self._validate_input(user_input)
+            if error:
+                errors["base"] = error
+            else:
+                return self.async_create_entry(
+                    title=user_input[CONF_HOST], data=user_input
+                )
+
+        return self.async_show_form(
+            step_id="edit", data_schema=self.data_schema, errors=errors
+        )
+
+    async def async_step_import(self, config):
+        """Import a config flow from configuration."""
+        _base_schema()(config)
```

Comment:
It looks like this is validating the schema here.  We should leave the platform schema for now since it will do the validation of the yaml which means we will always have valid data to import.

Reply:
Ok, will remove.